numpy/core/src/npymath/npy_math_internal.h.src

```diff
@@ -567,7 +567,8 @@ npy_divmod@c@(@type@ a, @type@ b, @type@ *modulus)
 
     /* adjust fmod result to conform to Python convention of remainder */
     if (mod) {
-        if (isless(b, (@type@)0) != isless(mod, (@type@)0)) {
+        if (!isnan@c@(b) && !isnan@c@(mod) && !isinf@c@(b) && !isinf@c@(mod) &&
+            ((b < (@type@)0) != (mod < (@type@)0))) {
```

Comment:
Just to get a start on this, how come this is necessary?  Is the compiler not C99 conforming or are some tests flake?
Note that we already ignore some warnings on certain platforms, so if it is the latter, it might also just be necessary to ignore the warnings.  (because in the end, few users care)

If it is not C99 compliant, I am not sure, I find that surprising (unless you are compiling with a C++ compiler rather C?)

Reply:
Sorry for the late reply.

When I do a regression test on NumPy built with fcc, it fails on `np.floor_divide` for float128.
In the current fujitsu compiler, when isgreater(nan, 1.0l) or isless(nan, 1.0l) is performed for float128, it compares the number with nan without checking nan and the `FE_INVALID` flag is raised. This is the cause of the error.
When doing `np.floor_divide` with Nan, the message `<stdin>:1: RuntimeWarning: invalid value encountered in floor_divide` appears.
- In the case of float128
```python
$ python
>>> import numpy as np
>>> fnan = np.array(np.nan, dtype='float128')
>>> fone = np.array(1.0, dtype='float128')
>>> np.floor_divide(fnan, fone)
<stdin>:1: RuntimeWarning: invalid value encountered in floor_divide
nan
```
- Other than float128
```python
$ python
>>> import numpy as np
>>> fnan = np.array(np.nan)
>>> fone = np.array(1.0)
>>> np.floor_divide(fnan, fone)
nan
```
Therefore, we have rewritten the code to check for Nan and run if Nan is not present.
If we change the code regarding this part, it will have a large impact on other environments. So, I think it is better not to change the code, but to make it a limitation when using the Fujitsu compiler. What do you think? I will change it so that regression tests that fail because of this will be skipped.

Reply:
Oh, just the longdouble implementation of `isless` gets it wrong!  Yes, I would suggest to skip the test, since it seems the most pragmatic choice.  This had been giving warnings for years on most platforms before we fixed it :).

We have some tests that have for example (I might prefer `xfail` really, but either works):
```
@pytest.mark.skipif(platform.machine() == "armv5tel",
                    reason="See gh-413.")
```

Reply:
Thank you very much. I had no idea that this had happened in the past on other platforms.

I would like to make sure that NumPy was built with fujitsu compiler as this problem only happens when built with this.

I don't feel like I can determine this from sys.platform, etc. Do you know any way to verify used compiler when doing pytest?

Reply:
Actually, not sure. @ganesh-k13 is there a simple way to get the compiler used?  Sounds like it would also be nice to include in `np.show_runtime()`.

Reply:
Thanks, but that means there is no easy way to get it right now to skip a test, I guess.  Maybe a very specific platform rule could work in practice?